libs/@local/hashql/core/src/type/environment.rs

```diff
@@ -0,0 +1,370 @@
+use core::ops::{Deref, DerefMut};
+use std::collections::HashMap;
+
+use super::{
+    Type, TypeId, TypeKind,
+    error::{TypeCheckDiagnostic, circular_type_reference},
+    generic_argument::GenericArgumentId,
+    recursion::RecursionBoundary,
+    unify_type_impl,
+};
+use crate::{
+    arena::transaction::{Checkpoint, TransactionalArena},
+    span::SpanId,
+};
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Default)]
+pub enum Variance {
+    #[default]
+    Covariant, // Same direction
+    Contravariant, // Opposite direction
+    Invariant,     // Exact match
```

Comment:
The comments should be doc-comments or at least be placed like one

Reply:
documented in [a4a7829](https://github.com/hashintel/hash/pull/6918/commits/a4a7829ca46e5f4a762eb74fe98f9b706c2e2780)